pkg/engine/validate/validate.go

```diff
@@ -137,12 +139,24 @@ func validateMap(log logr.Logger, resourceMap, patternMap map[string]interface{}
 		// - Equality
 		handler := anchor.CreateElementHandler(key, patternElement, path)
 		handlerPath, err := handler.Handle(validateResourceElement, resourceMap, origPattern, ac)
-		// if there are resource values at same level, then anchor acts as conditional instead of a strict check
-		// but if there are none then it's an if-then check
 		if err != nil {
-			// If global anchor fails then we don't process the resource
+			if skip(err) {
+				skipErrors = append(skipErrors, err)
+				continue
+			}
+
 			return handlerPath, err
 		}
+
+		applyCount++
+	}
+
+	if applyCount == 0 && len(skipErrors) > 0 {
+		return path, &PatternError{
```

Comment:
Does the condition here indicate that all rules are skipped (since it returns directly on non-skipped errors on L148)? Is this considered an `error` or `skip`?

Reply:
Yes, the idea here is to return skip only if all rules are skipped. This will considered an `skip`.
What I've done here actually references the logic in the `validateArrayOfMaps` function, as shown below:
https://github.com/kyverno/kyverno/blob/32160ba1cdc40918f3819c052e9c8a970d9992d4/pkg/engine/validate/validate.go#L232-L261

Reply:
Can we add one more test step to check the skip status in the final policy report? A reference `test/conformance/chainsaw/reports/admission/update/report-pass-assert.yaml` for example.

Reply:
Right now I've written two tests where one result is pass and the other result is fail. Do you mean I should write another test where all rules are skipped?

Reply:
Current two tests check the event, I'm saying add another test to check the `skip` result in the policy report explicitly.